### PR TITLE
[refacto] - refacto: agent popularity query

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the agent popularity query to improve efficiency and reduce database load. The main changes include:
- Removing the user count calculation, as it's no longer needed for the UI.
- Changing the time period for popularity calculation from 30 days to 7 days.
- Always show a modal when changing the scope of an assistant

## Risk

Overlooking files in which `userCount` was used.

There will be a discrepancy between in the message count until the Redis caches expire. Indeed, we are switching from 30 days to 7 days and what's currently in Redis cache is the count for 30 days.

## Deploy Plan

1. Test on `front-edge`
2. Deploy to `front`

### CODE REVIEW

File: front/lib/api/assistant/agent_usage.ts [hunk 1]
```diff
@@ -6,13 +6,12 @@ import {
...
   Message,
-  UserMessage,
 } from "@app/lib/models/assistant/conversation";
 import { Workspace } from "@app/lib/models/workspace";
 import { redisClient } from "@app/lib/redis";
 
-// Ranking of agents is done over a 30 days period.
-const rankingTimeframeSec = 60 * 60 * 24 * 30; // 30 days
+// Ranking of agents is done over a 7 days period.
+const rankingTimeframeSec = 60 * 60 * 24 * 7; // 30 days
```
Comment (philipperolet) [inline front/lib/api/assistant/agent_usage.ts:14]: nit: remove 30 days comment

**Review by philipperolet — APPROVED** [review]
LGTM :+1:

File: front/lib/api/assistant/agent_usage.ts (hunk 1 above)
Comment (JulesBelveze) [reply front/lib/api/assistant/agent_usage.ts]: Oops overlooked that one sorry 👍🏼

**Review by flvndvd — APPROVED** [review]
Thank you 🙇

**Comment by JulesBelveze** [comment]
@albandum The main reason we were storing usage on Redis is because those queries are really expensive and we do not want them to be ran for each admin landing on the workspace page.

**Comment by albandum** [comment]
@JulesBelveze looks like we removed that from the workspace page, now it only shows when you open an assistant modal if I'm not mistaken so might not be necessary anymore

**Pull request merged**